src/api/substring.cc

```diff
@@ -0,0 +1,170 @@
+/**
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
+* This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+**/
+#ifndef STRINGSUBSTRINGOBJECT_CC
+#define STRINGSUBSTRINGOBJECT_CC
+
+#include <node.h>
+#include <vector>
+
+#include "substring.h"
+#include "../iastlimits.h"
+#include "../tainted/range.h"
+#include "../tainted/transaction.h"
+#include "../tainted/string_resource.h"
+#include "../iast.h"
+#include "../utils/propagation.h"
+#include "v8.h"
+
+#define TO_V8STRING(arg) (v8::Local<v8::String>::Cast(arg))
+#define TO_INTEGER_VALUE(arg, v8_ctx) (arg->IntegerValue(v8_ctx).FromJust())
+
+using v8::Exception;
+using v8::FunctionCallbackInfo;
+using v8::Isolate;
+using v8::Local;
+using v8::NewStringType;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+using iast::utils::GetLocalStringPointer;
+using iast::utils::getRangesInSlice;
+
+namespace iast {
+namespace api {
+void substring(const FunctionCallbackInfo<Value>& args) {
+    Isolate* isolate = args.GetIsolate();
+    auto context = isolate->GetCurrentContext();
+    int argc = args.Length();
+
+    if (argc < 4) {
+        isolate->ThrowException(Exception::TypeError(
```

Comment:
Umm, no Error is thrown when you call `substring()` with no arguments. Maybe here we should return the `result`argument?